src/daft-dsl/src/python.rs

```diff
@@ -187,9 +199,19 @@ pub fn stateful_udf(
     return_dtype: PyDataType,
     resource_request: Option<ResourceRequest>,
     init_args: Option<&PyAny>,
+    batch_size: Option<i64>,
```

Comment:
Would using `usize` or `u32` across the codebase be more appropriate?

Reply:
Yep, didn't know that PyO3 would automatically error on negatives